### PR TITLE
[BEAM-2785] Allow termination of DirectRunner execution with Ctrl-C

### PR DESCRIPTION
**Pull request by charlesccychen**
R: @aaltay

### CODE REVIEW

File: sdks/python/apache_beam/runners/direct/executor.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def __init__(self, queue, index):
       self._default_name = 'ExecutorServiceWorker-' + str(index)
       self._update_name()
       self.shutdown_requested = False
+
+      # Stop worker thread when main thread exists.
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/direct/executor.py]: `exits`

**Comment by charlesccychen** [comment]
Thanks, please take another look.

**Comment by aaltay** [comment]
LGTM, will merge after tests.

**Comment by charlesccychen** [comment]
retest this please

**Comment by charlesccychen** [comment]
Note that after this change, it makes less sense to use the DirectRunner without .wait_until_finish(), as termination of the main thread will implicitly terminate any running pipelines.

**Comment by aaltay** [comment]
> as termination of the main thread will implicitly terminate any running pipelines.

This is different compared to cluster based runners, however I think it makes sense. Do you think we should have a log message if we can detect a pre-mature termination with `DirectRunner`?

**Comment by charlesccychen** [comment]
Thanks, I have added the suggested warning.

**Comment by aaltay** [comment]
LGTM. Thank you.

**Comment by aaltay** [comment]
Retest this please

**Pull request closed without merging**